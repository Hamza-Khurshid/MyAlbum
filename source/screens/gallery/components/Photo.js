import React from 'react'
import styles from './style'
import { View, Image, Text } from 'react-native'

export default function Photo({ photo, album }) {
    return (
        <View style={styles.photoContainer}>
            <Image source={{ uri: photo.url }} style={styles.photo} />
            <Text style={styles.photoTitle}>{photo.title}</Text>
            <Text style={styles.albumTitle}>{album}</Text>
        </View>
    )
}
