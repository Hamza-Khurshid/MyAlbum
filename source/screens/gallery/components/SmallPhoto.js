import React from 'react'
import styles from './style'
import { Image, TouchableOpacity } from 'react-native'

export default function SmallPhoto({ photo, index, activeIndex, setActivePos }) {
    return (
        <TouchableOpacity onPress={()=>setActivePos(index)}>
            <Image
                source={{ uri: photo.url }}
                style={[styles.smallPhoto, activeIndex === index ? styles.activePhoto : {}]}
            />
        </TouchableOpacity>
    )
}
