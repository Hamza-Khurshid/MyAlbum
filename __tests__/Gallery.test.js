import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Photo from '../source/screens/gallery/components/Photo';
import SmallPhoto from '../source/screens/gallery/components/SmallPhoto';

test('Photo renders correctly', () => {
    const tree = renderer.create(<Photo photo={{ url: "", title: "" }} />).toJSON();
    expect(tree).toMatchSnapshot();
});


test('SmallPhoto renders correctly', () => {
    let photo = { url: "", title: "" }
    let index = 0;
    let activeIndex = 0;
    let setActivePos = () => {}

    const tree = renderer.create(<SmallPhoto photo={photo} index={index} activeIndex={activeIndex} setActivePos={setActivePos} />).toJSON();
    expect(tree).toMatchSnapshot();
});