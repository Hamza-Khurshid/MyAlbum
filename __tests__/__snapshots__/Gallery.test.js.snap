// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photo renders correctly 1`] = `
<View
  style={
    Object {
      "height": 1334,
      "width": 750,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "",
      }
    }
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <Text
    style={
      Object {
        "bottom": 200,
        "color": "white",
        "fontSize": 16,
        "position": "absolute",
        "textAlign": "center",
        "width": "100%",
      }
    }
  >
    
  </Text>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 22,
        "paddingHorizontal": 10,
        "position": "absolute",
        "textAlign": "center",
        "top": 40,
        "width": "100%",
      }
    }
  />
</View>
`;

exports[`SmallPhoto renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "",
      }
    }
    style={
      Array [
        Object {
          "borderColor": "gray",
          "borderRadius": 10,
          "borderWidth": 1,
          "height": 80,
          "marginRight": 10,
          "width": 80,
        },
        Object {
          "borderColor": "white",
          "borderWidth": 2,
        },
      ]
    }
  />
</View>
`;
